core/mutator/entry/mutator.go

```diff
@@ -43,6 +44,46 @@ func MapLogItemFn(m *mutator.LogMessage, emit func(index []byte, mutation *pb.En
 	return nil
 }
 
+// ReduceFn decides which of multiple updates can be applied in this revision.
+func ReduceFn(index []byte, msgs []*pb.EntryUpdate, leaves []*tpb.MapLeaf, emit func(*tpb.MapLeaf)) error {
+	if got := len(leaves); got > 1 {
+		return fmt.Errorf("expected 0 or 1 map leaf for index %x, got %v", index, got)
+	}
+	var oldValue *pb.SignedEntry // If no map leaf was found, oldValue will be nil.
+	var err error
+	if len(leaves) > 0 {
+		oldValue, err = FromLeafValue(leaves[0].GetLeafValue())
+		if err != nil {
+			return fmt.Errorf("entry.FromLeafValue(): %v", err)
+		}
+	}
+
+	if len(msgs) == 0 {
+		return fmt.Errorf("no msgs for index %x", index)
+	}
+
+	// TODO(gbelvin): Choose the mutation deterministically, regardless of the messages order.
```

Comment:
This TODO seems higher priority now that you've explicitly stated on the interface that it needs to be associative.

Reply:
Fixed